### ISSUE TITLE
Selective validation of hidden selects, decorated with jquery selectmenu

### ISSUE THREAD

**Issue by alexandis** [jzaefferer/jquery-validation#1738]
I use jquery selectmenu to decorate my HTML select elements.
Also I use jquery validation.

The question is: how to DIFFER between select elements, that are just in hidden area of page (and thus really should not be validated) and select elements, that should be validated (the ones, that are in visible area, but hidden by selectmenu)?
Currently I just set
`$(instance.form).validate().settings.ignore = [];`
in page edit mode
and when edit is finished I set
`$(instance.form).validate().settings.ignore = ':hidden';`
back.

I make selective validation of the part of page by using such method:

`	function isFormSubsetValid($root) {
		var handlerInstance = this;
		var $validator = $.data($(handlerInstance.instance.form).get(0), 'validator');
		var isValid = true;
		$root.filter(':input').add($root.find(':input')).each(function () {
			if (!$validator.element($(this))) {
				isValid = false;
				return false;
			}
		});
		return isValid;
	};
`
However, this works only as long as the area I check ($root), does not include hidden part, that should not be checked, right?

**Comment by Arkni** [comment]
> The question is: how to DIFFER between select elements, that are just in hidden area of page (and thus really should not be validated) and select elements, that should be validated (the ones, that are in visible area, but hidden by selectmenu)?

Add a `class` to the select elements that you want to ignore and then use:
```js
ignore: "not(.your_class)"
```

**Comment by alexandis** [comment]
Thanks. Knew about this feature. However, what if it's near impossible to assign some classes?
Is it possible to play on the following difference:
- I have invisible subarea of area I validate: all elements inside, including selects, should be ignored;
- I have visible subarea of the same area: all selects inside, even invisible, should be validated
?

**Comment by staabm** [comment]
You can ignore whatever you are able to describe with a css selector

**Comment by alexandis** [comment]
Ok, got it. Everything is ruled by selectors only. Thanks.

**Comment by staabm** [comment]
Does this work in your use-case?

**Comment by alexandis** [comment]
Sorry to bother you again, i've just tried different approaches to make it work, but reopen the issue asking for more help. The matter is i have a wide area of zones with elements: i get some chunk of html with AJAX and want to set only this specific markup to be validated (jquery dialog body, etc). I hoped i could use parent selector and all elements within it would be validated. But now i understand I need to specify EACH element's selector. Getting pretty complex in my case. Is there other way to resolve my issue?

**Comment by staabm** [comment]
its hard to follow your description. could you post a simplified jsfiddle which demonstrates your problem?

**Comment by alexandis** [comment]
I think I will close the issue now, because now everything seems to be resolved. Thanks. Maybe I would be able to help to improve plugin in some way, but I'm really lack of time a lot. Sorry.
